op-challenger/fault/responder.go

```diff
@@ -0,0 +1,100 @@
+package fault
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// faultResponder implements the [Responder] interface to send onchain transactions.
+type faultResponder struct {
+	log log.Logger
+
+	txMgr txmgr.TxManager
+
+	fdgAddr common.Address
+	fdgAbi  *abi.ABI
+}
+
+// NewFaultResponder returns a new [faultResponder].
+func NewFaultResponder(logger log.Logger, txManagr txmgr.TxManager, fdgAddr common.Address) (*faultResponder, error) {
+	fdgAbi, err := bindings.FaultDisputeGameMetaData.GetAbi()
+	if err != nil {
+		return nil, err
+	}
+	return &faultResponder{
+		log:     logger,
+		txMgr:   txManagr,
+		fdgAddr: fdgAddr,
+		fdgAbi:  fdgAbi,
+	}, nil
+}
+
+// buildFaultDefendData creates the transaction data for the Defend function.
+func (r *faultResponder) buildFaultDefendData(parentContractIndex int, pivot [32]byte) ([]byte, error) {
+	return r.fdgAbi.Pack(
+		"defend",
+		big.NewInt(int64(parentContractIndex)),
+		pivot,
+	)
+}
+
+// buildFaultAttackData creates the transaction data for the Attack function.
+func (r *faultResponder) buildFaultAttackData(parentContractIndex int, pivot [32]byte) ([]byte, error) {
+	return r.fdgAbi.Pack(
+		"attack",
+		big.NewInt(int64(parentContractIndex)),
+		pivot,
+	)
+}
+
+// buildTx builds the transaction for the [faultResponder].
+func (r *faultResponder) buildTx(ctx context.Context, response Claim) ([]byte, error) {
```

Comment:
IMO this is the most important function to test. If we test that it's calling to the mock send function, we should also be asserting that the data it is sending is correct, not just that it made the correct number of calls.